swaybar/ipc.c

```diff
@@ -305,16 +305,18 @@ bool ipc_get_workspaces(struct swaybar *bar) {
 				ws->num = json_object_get_int(num);
 				ws->name = strdup(json_object_get_string(name));
 				ws->label = strdup(ws->name);
-				size_t len_offset = ws->name[1] == ':' ? 2 : 1;
+				size_t len_offset = numlen(ws->num);
```

Comment:
This can do weird stuff if `ws->num == -1`.

i3 also strips the colon if the name is something like `1:`. I don't think this does that, though admittedly I haven't actually tested it.

Reply:
Nice catch with `numlen`, I have modified it to pass the absolute value to `log10`. Hanging colon will be stripped now too.